src/controllers/paymentController.ts

```diff
@@ -0,0 +1,58 @@
+import { Request, Response } from 'express'
+import Stripe from 'stripe'
+import Order, { OrderStatus } from '../database/models/orderModel'
+import OrderService from '../services/orderService'
+import {STRIPE_SECRET_KEY} from '../config/index'
+
+const stripe = new Stripe(STRIPE_SECRET_KEY, {
+  apiVersion: '2024-04-10',
+})
+/**
+ * payment Controller class
+ */
+export default class PaymentController {
+  /**
+   * Process payment and update order status
+   * @param {Request} req - Express request object
+   * @param {Response} res - Express response object
+   * @returns {Promise<Response>} Promise that resolves to an Express response
+   */
+  static async processPayment(req: Request, res: Response): Promise<Response> {
+    const { currency, paymentMethodId } = req.body
+    const orderId = req.params.orderId
+    const userId = req.user?.id
+
+    if (!userId) {
+      return res.status(400).json({ message: 'User ID is required' })
+    }
+
+    const order = await Order.findByPk(orderId)
+    if (!order || order.userId !== userId) {
+      return res.status(404).json({ message: 'Order not found' })
+    }
+
+    const amount = order.totalAmount
+
+    const paymentIntent = await stripe.paymentIntents.create({
+      amount,
+      currency,
+      payment_method: paymentMethodId,
+      confirm: true,
+      metadata: { orderId, userId },
+      return_url: 'https://example.com/success',
+    })
+
+    const updatedOrder = await OrderService.updateOrderStatus(
+      orderId,
+      OrderStatus.CONFIRMED,
+    )

```

Comment:
```suggestion
    const updatedOrder = await OrderService.updateOrderStatus(
      orderId,
      OrderStatus.INIATED,
    )

```
can you check that we don't have the callback from the stripe? if we have the callback you can update make OrderStatus to (iniated or processing )and then you can create  another endpoint to receive callback and update order to the status from callback, sometimes it might be failed or success